intents-lab/app/src/main/java/com/owino/intents_lab/services/DownloadService.kt

```diff
@@ -1,22 +1,27 @@
 package com.owino.intents_lab.services
 
-import android.app.Service
 import android.content.Intent
 import android.os.Binder
 import android.os.IBinder
 import android.widget.Toast
+import androidx.lifecycle.LifecycleService
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import java.net.HttpURLConnection
 import java.net.URL
 import java.nio.charset.Charset
 
-class DownloadService : Service() {
+class DownloadService : LifecycleService() {
```

Comment:
LifecycleService seems more promising, I just wonder if you can still launch it with ```startService(intent)```?

Reply:
Yes absolutely!